### PR TITLE
fix: handling infinite loop

### PR DESCRIPTION
**Pull request by harshilgajera-crest**
There was a bug in PSA where if in splunk_ingest_data first worker errors out or throws exception then it does not write to a file, now other workers would be waiting indefinitely for the file to be present, So the execution would continue in infinitly.

To handle this scenario, we can handle the exception thrown by the first worker and make sure that in any case it writes to a file, so that other workers don't wait indefinitly

### CODE REVIEW

**Comment by mkolasinski-splunk** [comment]
@harshilgajera-crest can you switch base to develop, so we first get beta release and run some testing in CI?

**Review by artemrys — CHANGES_REQUESTED** [review]

File: .github/workflows/build-test-release.yml [hunk 1]
```diff
@@ -120,18 +120,15 @@ jobs:
...
-          export SPLUNK_ADDON=TA_fiction
-          export SPLUNK_APP_ID=TA_fiction
+          export SPLUNK_APP_PACKAGE=./tests/e2e/addons/TA_fiction_indextime
+          export SPLUNK_ADDON=TA_fiction_indextime
+          export SPLUNK_APP_ID=TA_fiction_indextime
           export SPLUNK_VERSION=${{ matrix.splunk.version }}
+          export SPLUNK_HEC_TOKEN="9b741d03-43e9-4164-908b-e09102327d22"
           echo $SPLUNK_VERSION
           docker compose -f "docker-compose-ci.yml" build
-          SPLUNK_PASSWORD=Chang3d! docker compose -f docker-compose-ci.yml up -d splunk
```
[the remark this answers is not in the document]
Comment (harshilgajera-crest) [reply .github/workflows/build-test-release.yml]: We are not actually removing this, 
This line SPLUNK_PASSWORD=Chang3d! docker compose -f docker-compose-ci.yml up --abort-on-container-exit
would do the same thing, it would start all the services, earlier we did in individually which seems to be not required.

File: tests/e2e/test_splunk_addon.py [hunk 2]
```diff
@@ -737,5 +737,50 @@ def empty_method():
...
+        "--splunk-host=splunk",
+        "--splunk-data-generator=tests/addons/TA_fiction_indextime/default",
+        "--sc4s-host=splunk",
+        "--sc4s-port=100",
+        "-n 2",
+    )
+
+    # Here we are not interested in the failures or errors,
+    # we are basically checking that we get results and test execution does not get stuck
+    assert result.parseoutcomes().get("passed") > 0
```
[the remark this answers is not in the document]
Comment (harshilgajera-crest) [reply tests/e2e/test_splunk_addon.py:786]: I don't think so, because that is handled internally via xdist. But everything would run on either of the workers.

File: tests/e2e/test_splunk_addon.py [hunk 3]
```diff
@@ -737,5 +737,50 @@ def empty_method():
...
+        os.path.join(testdir.request.fspath.dirname, "test_data_models"),
+        os.path.join(testdir.tmpdir, "tests/data_models"),
+    )
+
+    setup_test_dir(testdir)
+    SampleGenerator.clean_samples()
+    Rule.clean_rules()
+
+    # run pytest with the following cmd args
+    # we are providing wrong sc4s service details here so that we can recreate scenario where first worked raises execption and other workers get stuck
```
[the remark this answers is not in the document]
Comment (harshilgajera-crest) [reply tests/e2e/test_splunk_addon.py]: Done

File: .github/workflows/build-test-release.yml [hunk 4]
```diff
@@ -120,18 +120,15 @@ jobs:
         run: |
           pip install git+https://github.com/pixelb/crudini
           mkdir test-results-${{ matrix.splunk.version }}
-      - name: Splunk Up
+      - name: Test
         run: |
-          export SPLUNK_APP_PACKAGE=./tests/e2e/addons/TA_fiction
-          export SPLUNK_ADDON=TA_fiction
-          export SPLUNK_APP_ID=TA_fiction
+          export SPLUNK_APP_PACKAGE=./tests/e2e/addons/TA_fiction_indextime
```
Comment (mkolasinski-splunk) [inline .github/workflows/build-test-release.yml:125]: What is the reason of this change?

File: tests/e2e/test_splunk_addon.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (artemrys) [reply tests/e2e/test_splunk_addon.py:786]: What about checking the logs if those are available somewhere?

My point is - if you don't make a fix - will this test pass anyways?

[the remark this answers is not in the document]
Comment (harshilgajera-crest) [reply tests/e2e/test_splunk_addon.py:786]: Okay, So I actually tested it on main branch without making the fix and test kept on running.
Let me make the same in github actions, I will provide a link here for the actions run which should go on for 6hrs.

Here is the link with test change in main branch without fix: https://github.com/splunk/pytest-splunk-addon/actions/runs/8753677154/job/24023897918

This will go on forever till runner timeout.

**Review by mkolasinski-splunk — APPROVED** [review]

**Review by kdoroszko-splunk — APPROVED** [review]
